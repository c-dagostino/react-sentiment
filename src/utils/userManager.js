import { createUserManager } from 'redux-oidc';
import { WebStorageStateStore, InMemoryWebStorage } from 'oidc-client';

const {
  REACT_APP_OIDC_AUTHORITY_CLIENT_ID: clientId,
  REACT_APP_OIDC_AUTHORITY_URL: authorityUrl
} = process.env;

const userManagerConfig = {
  client_id: clientId,
  redirect_uri: `${window.location.protocol}//${window.location.hostname}${
    window.location.port ? `:${window.location.port}` : ''
  }/callback`,
  response_type: 'token id_token',
  scope: 'openid email name',
  authority: authorityUrl,
  silent_redirect_uri: `${window.location.protocol}//${
    window.location.hostname
  }${window.location.port ? `:${window.location.port}` : ''}/silent_renew.html`,
  automaticSilentRenew: true,
  filterProtocolClaims: true,
  loadUserInfo: true
};

if (process.env.NODE_ENV === 'test') {
  const stateStoreStorage = new InMemoryWebStorage();
  const userStoreStorage = new InMemoryWebStorage();

  userManagerConfig.stateStore = new WebStorageStateStore({
    store: stateStoreStorage
  });
  userManagerConfig.userStore = new WebStorageStateStore({
    store: userStoreStorage
  });
}

// we add the localStorage so we can share the user between tabs
if (window.localStorage) {
  userManagerConfig.userStore = new WebStorageStateStore({
    store: window.localStorage
  });
}

const userManager = createUserManager(userManagerConfig);
export default userManager;
