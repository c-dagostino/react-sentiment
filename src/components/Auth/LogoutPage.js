import React from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';

import userManager from '../../utils/userManager';
import { LOGOUT_URL } from './constants';

const { REACT_APP_OIDC_AUTHORITY_URL: authorityUrl } = process.env;

class LogoutPage extends React.Component {
  static propTypes = {
    history: PropTypes.shape({
      push: PropTypes.func.isRequired
    }).isRequired
  };

  async componentDidMount() {
    try {
      const user = await userManager.getUser();
      const tokenId = user.id_token;

      userManager.removeUser();
      window.location.assign(
        `${authorityUrl}${LOGOUT_URL}&id_token_hint=${tokenId}`
      );
    } catch (exception) {
      const { push } = this.props.history;
      return push('/');
    }
    return null;
  }

  render() {
    return <div>Logging out...</div>;
  }
}

export default withRouter(LogoutPage);
