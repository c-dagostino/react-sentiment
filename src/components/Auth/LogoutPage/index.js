import React from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';

import { FormattedMessage } from 'react-intl';

import userManager from '../../../utils/userManager';
import { LOGOUT_URL } from '../constants';

const { REACT_APP_OIDC_AUTHORITY_URL: authorityUrl } = process.env;

export class LogoutPage extends React.Component {
  static propTypes = {
    history: PropTypes.shape({
      push: PropTypes.func.isRequired
    }).isRequired
  };

  // eslint-disable-next-line consistent-return
  async componentDidMount() {
    try {
      const user = await userManager.getUser();
      const tokenId = user.id_token;

      await userManager.removeUser();
      window.location.assign(
        `${authorityUrl}${LOGOUT_URL}&id_token_hint=${tokenId}`
      );
    } catch (exception) {
      const { push } = this.props.history;
      return push('/');
    }
  }

  render() {
    return (
      <FormattedMessage
        id={'logoutPage.title'}
        defaultMessage={'[LogoutPage]'}
      />
    );
  }
}

export default withRouter(LogoutPage);
