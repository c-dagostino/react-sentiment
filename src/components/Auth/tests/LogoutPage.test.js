import React from 'react';
import { shallow } from 'enzyme';
import { LogoutPage } from '../LogoutPage';
import userManager from '../../../utils/userManager';

function getWrapper(props) {
  const requiredProps = {
    history: {
      push: () => {}
    }
  };
  return shallow(<LogoutPage {...requiredProps} {...props} />);
}

describe('LogoutPage', () => {
  it('Expect component to render properly', async () => {
    const mockGetUser = jest
      .spyOn(userManager, 'getUser')
      .mockImplementation(() => Promise.resolve({ id_token: 'token-test' }));
    const wrapper = getWrapper();
    expect(wrapper).toMatchSnapshot();
    expect(mockGetUser).toHaveBeenCalled();
  });

  it('Expect component to redirect on error', async () => {
    jest.spyOn(userManager, 'getUser').mockImplementation(() => {});
    const wrapper = getWrapper();
    expect(wrapper).toMatchSnapshot();
  });
});
