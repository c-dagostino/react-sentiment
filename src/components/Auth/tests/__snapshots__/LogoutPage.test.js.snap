// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogoutPage Expect component to redirect on error 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LogoutPage
    history={
        Object {
            "push": [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Logging out...",
    },
    "ref": null,
    "rendered": "Logging out...",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Logging out...",
      },
      "ref": null,
      "rendered": "Logging out...",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`LogoutPage Expect component to render properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LogoutPage
    history={
        Object {
            "push": [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Logging out...",
    },
    "ref": null,
    "rendered": "Logging out...",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Logging out...",
      },
      "ref": null,
      "rendered": "Logging out...",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
